### PR TITLE
feat(modal): customization

### PR DESCRIPTION
**Pull request by andipants12**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by nkrantz — COMMENTED** [review]
It's looking great so far! Missing tests and changeset but component names are good.

File: packages/paste-core/components/modal/src/Modal.tsx [hunk 1]
```diff
@@ -123,22 +129,30 @@ const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
         {transitions(
           (styles, item) =>
             item && (
-              <ModalDialogOverlay
+              <Box
+                // @ts-expect-error Render overlay as box for customization
```
Comment (SiTaggart) [inline packages/paste-core/components/modal/src/Modal.tsx:133]: I actually don't _mind_ this approach, but if there is some push back you can achieve it with styled components. We did it in Textarea https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/textarea/src/TextArea.tsx#L71

The custom css function is exported from Box, and you can pass it to the styled component manually to merge in the custom styles from `element`. You just need to manually apply the `data-paste-element` attributes on the node you want to target. 

I don't have a preference on either way. This is just knowledge share

File: packages/paste-core/components/modal/src/Modal.tsx [hunk 2]
```diff
@@ -147,16 +161,14 @@ const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
...
-  };
-}
+Modal.propTypes = {
+  children: PropTypes.node.isRequired,
+  isOpen: PropTypes.bool.isRequired,
+  onDismiss: PropTypes.func.isRequired,
+  allowPinchZoom: PropTypes.bool,
+  size: PropTypes.oneOf(['default', 'wide'] as Sizes[]).isRequired,
+  initialFocusRef: PropTypes.object as any,
+  ariaLabelledby: PropTypes.string.isRequired,
```
Comment (SiTaggart) [inline packages/paste-core/components/modal/src/Modal.tsx:171]: Should we be adding `element` to all these propTypes, across all the components in the PR?

Comment (andipants12) [reply packages/paste-core/components/modal/src/Modal.tsx:171]: Ah yes we should, that was an oversight. Will add it now. 👍

File: packages/paste-core/components/modal/__tests__/customization.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    (
+      {
+        children,
+        'data-paste-element': dataPasteElement,
+        style,
+        className,
+      }: {children: any; 'data-paste-element': string; style: any; className: string},
+      ref: any
+    ) => (
+      <div
```
Comment (andipants12) [inline packages/paste-core/components/modal/__tests__/customization.spec.tsx:21]: Mocking the react portal for this test.

**Review by nkrantz — APPROVED** [review]
I like the helper functions you made! This is looking good 👍

File: packages/paste-core/components/modal/__tests__/customization.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+import {render, screen} from '@testing-library/react';
+import {CustomizationProvider} from '@twilio-paste/customization';
+import {matchers} from 'jest-emotion';
+
+import {BaseModal, initStyles} from '../stories/customization.stories';
+
+expect.extend(matchers);
+
+jest.mock('@twilio-paste/modal-dialog-primitive', () => {
+  const actual = jest.requireActual('@twilio-paste/modal-dialog-primitive');
```
Comment (SiTaggart) [inline packages/paste-core/components/modal/__tests__/customization.spec.tsx:12]: I guess I'm a little unsure on why this needs to be mocked, so providing some sort of commentary on the why might be helpful in a code comment.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**